reader.lua

```diff
@@ -245,76 +245,81 @@ end
-    if file and file ~= "" then
-        local ReaderUI = require("apps/reader/readerui")
-        UIManager:nextTick(function()
-            ReaderUI:showReader(file)
-        end)
-    -- we assume a directory is given in command line argument
-    -- the filemanger will show the files in that path
+end
+-- if file is given in command line argument the given file is opened in the reader.
+if file and file ~= "" then
+    local ReaderUI = require("apps/reader/readerui")
+    UIManager:nextTick(function()
+        ReaderUI:showReader(file)
+    end)
+    exit_code = UIManager:run()
+-- if the quickstart was not shown, show it.
+else
+    local QuickStart = require("ui/quickstart")
+    if not QuickStart:isShown() then
+        start_with = "last"
+        last_file = QuickStart:getQuickStart()
+    end
+end
+-- if start_ with is last document makes sure it exists.
+if start_with == "last" and last_file then
+    if lfs.attributes(last_file, "mode") ~= "file" then
+        UIManager:show(retryLastFile())
     else
-        local FileManager = require("apps/filemanager/filemanager")
-        local home_dir =
-            G_reader_settings:readSetting("home_dir") or ARGV[argidx]
+        local ReaderUI = require("apps/reader/readerui")
         UIManager:nextTick(function()
-            FileManager:setRotationMode(true)
-            FileManager:showFiles(home_dir)
+            ReaderUI:showReader(last_file)
         end)
-        -- always open history on top of filemanager so closing history
-        -- doesn't result in exit
-        if start_with == "history" then
-            local FileManagerHistory = require("apps/filemanager/filemanagerhistory")
-            UIManager:nextTick(function()
-                FileManagerHistory:onShowHist(last_file)
-            end)
-        elseif start_with == "favorites" then
-            local FileManagerCollection = require("apps/filemanager/filemanagercollection")
-            UIManager:nextTick(function()
-                FileManagerCollection:new{
-                    ui = FileManager.instance,
-                }:onShowColl("favorites")
-            end)
-        elseif start_with == "folder_shortcuts" then
-            local FileManagerShortcuts = require("apps/filemanager/filemanagershortcuts")
-            UIManager:nextTick(function()
-                FileManagerShortcuts:new{
-                    ui = FileManager.instance,
-                }:onShowFolderShortcutsDialog()
-            end)
-        end
     end
     exit_code = UIManager:run()
-elseif last_file then
-    local ReaderUI = require("apps/reader/readerui")
+-- or else we assume a directory is given in command line argument
+-- the filemanger will show the files in that path
+-- or the home path if start_with is defined.
+elseif (directory and directory ~= "") or start_with ~= nil then
+    local FileManager = require("apps/filemanager/filemanager")
+    local home_dir =
+        G_reader_settings:readSetting("home_dir") or directory
```

Comment:
There's also some kind of platform default dir stuff; is that still properly functional with this?

Reply:
That's a fallback to **go to home** that matches `Device.home_dir`

As is now directory is always passed as an argument on all platforms. That's why `G_reader_settings:readSetting("home_dir")` takes precedence over the directory.

It should be possible to rely on `Device.home_dir` to start the filemanager on the right path.

Reply:
> As is now directory is always passed as an argument on all platforms. 
Besides kindle, hence the source of this bug. :smile: 

That's why `G_reader_settings:readSetting("home_dir")` takes precedence over the directory.
> 
> It should be possible to rely on `Device.home_dir` to start the filemanager on the right path.